Sources/MapboxDirections/RouteStep.swift

```diff
@@ -776,6 +828,13 @@ open class RouteStep: Codable {
     */
     public let intersections: [Intersection]?
     
+    /**
+     An array of `Administrative Regions` indicies for each `intersection` along current step.
+     
+     Array may be `nil` in case `intersections` data is not available. Array element may be `nil` if corresponding `intersection` has no `Administrative Region` assigned.
+     */
+    public let administrativeRegionIndicesByIntersection: [Int?]?
```

Comment:
Per https://github.com/mapbox/mapbox-directions-swift/issues/477#issue-732531128, even though it’s desirable to round-trip these indices, the RouteStep API should publicly expose a `regionCodesByIntersection` property of type `[String]`. There’s no need to explicitly give developers an option between alpha-2 and alpha-3 codes as the API does, though I suppose this method can remain in place as a more indirect option.

Reply:
`RouteLeg.administrativeRegionIndicesByIntersection`, `RouteLeg.Intersections` and `Intersection.regionCode` are all constants. Is it possible to get them out of sync? Having `administrativeRegionIndicesByIntersection` as a computed property seem to become expensive getter.

Reply:
I guess I had been thinking these properties would need to be non-constant in order for the populate-after-decoding idea in https://github.com/mapbox/mapbox-directions-swift/pull/485#discussion_r519562906 to work. It’s still suboptimal for derived values to be stored explicitly, but you’re right that it’s even more suboptimal to make it a computed property over the whole step, since client code may only ever need one index at a time.

Mainly I was hoping to keep client code from having to subscript any array to find out the administrative region that contains a given intersection, but I guess it’s inevitable that indices will be part of this API no matter what. We could keep what you have so far, but add a `RouteLeg.regionCode(atStepIndex:intersectionIndex:)` method. At least that method will be performant by only looking for a single intersection, and developers could get the answer they’re looking for in one method call.

Reply:
Added the proposed convenience method